src/GGTestUtils.py

```diff
@@ -99,6 +104,23 @@ def _check_greengrass_group_deployment_status(self, deployment_id):
                 print(f"An error occurred: {e}")
             return None
 
+    def create_local_deployment(self, artifacts_dir, recipe_dir, component_details):
+        cli_cmd = ["sudo", self.get_cli_bin_path(), "deploy"]
+        if artifacts_dir is not None:
+            cli_cmd.extend(["--artifacts-dir", artifacts_dir])
+        if recipe_dir is not None:
+            cli_cmd.extend(["--recipe-dir", recipe_dir])
+        cli_cmd.append(f"--add-component={component_details}")
+        process = subprocess.run(cli_cmd, capture_output=True, text=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)